scripts/warp_2d_script.py

```diff
@@ -316,35 +317,37 @@ def beam_dens_func(x, y, z):
 # Continuously inject the plasma, if the moving window is on
 if use_moving_window :
     installuserinjection( plasma_injector.continuous_injection )
-        
+
 # Setup the diagnostics
 # ---------------------
 if write_fields == 1:
     diag1 = FieldDiagnostic( period=diag_period, top=top, w3d=w3d, em=em,
                 comm_world=comm_world, lparallel_output=parallel_output,
-                write_dir='./example-2d' )
-    installafterstep( diag1.write )
+                write_dir='./example-2d', fieldtypes=["E", "rho"] )
 if write_particles == 1:
     species_dict = { species.name : species for species in listofallspecies \
-            if not( species.name in ["Hydrogen0+", "electron from Hydrogen"] )}
+            if species.name == "electrons" }
     diag2 = ParticleDiagnostic( period=diag_period, top=top, w3d=w3d,
             species=species_dict, write_dir='./example-2d',
+            particle_data={"position","momentum","weighting","id"},
+            comm_world=comm_world, lparallel_output=parallel_output )
+    species_dict = { species.name : species for species in listofallspecies \
+            if species.name == "Hydrogen1+" }
+    diag3 = ParticleDiagnostic( period=diag_period, top=top, w3d=w3d,
+            species=species_dict, write_dir='./example-2d',
+            particle_data={"position"},
```

Comment:
can you please add `momentum` and `weighting` since those are required attributes in the `ED-PIC` extension to describe a minimum set of a particle species? (validator fails.)